File: Graph/Basic/Representation/WeightedAdjList/WeightedAdjList.cpp
#include<bits/stdc++.h>
using namespace std;
vector<vector<pair<int, int>>> printGraph(int V, vector<vector<int>> &edges) {
    vector<vector<pair<int, int>>> graph(V);
    for(auto edge: edges) {
        graph[edge[0]].push_back(make_pair(edge[1], edge[2]));
        graph[edge[1]].push_back(make_pair(edge[0], edge[2]));
    }
    return graph;
}
int main(void) {
    vector<vector<int>> edges;
    edges.push_back({0, 1, 3});
    edges.push_back({0, 4, 2});
    edges.push_back({4, 1, 5});
    edges.push_back({4, 3, 3});
    edges.push_back({1, 3, 6});
    edges.push_back({1, 2, 3});
    edges.push_back({3, 2, 5});
    vector<vector<pair<int, int>>> graph = printGraph(5, edges);
    for(auto i: graph) {
        for(pair<int, int> j: i) {
            cout << "(" << j.first << ", " << j.second << ") ";
        }
        cout << "\n";
    }
    return 0;
}

File: Graph/Basic/Representation/WeightedAdjList/WeightedAdjList.java

import java.util.ArrayList;
import java.util.List;

public class WeightedAdjList {
    static List<List<List<Integer>>> printGraph(int V, int [][]edges) {
        List<List<List<Integer>>> graph = new ArrayList<>(V);
        for(int i=0; i<V; i++) {
            graph.add(new ArrayList<>());
        }
        for(int[] edge: edges) {
            List<Integer> node1 = new ArrayList<>();
            node1.add(edge[1]);
            node1.add(edge[2]);
            graph.get(edge[0]).add(node1);
            List<Integer> node2 = new ArrayList<>();
            node2.add(edge[0]);
            node2.add(edge[2]);
            graph.get(edge[1]).add(node2);
        }
        return graph;
    }
    public static void main(String[] args) {
        int[][] edges = {{0, 1, 2}, {0, 4, 3}, {4, 1, 5}, {4, 3, 2}, {1, 3, 6}, {1, 2, 1}, {3, 2, 4}};
        List<List<List<Integer>>> list = printGraph(5, edges);
        for(List<List<Integer>> arr: list) {
            System.out.println(arr);
        }
    }
}
